lib/stages/applyGains.cpp

```diff
@@ -363,7 +376,15 @@ void applyGains::fetch_thread() {
         }
 
         // update gains
-        state_id_t state_id = dm.create_state<gainState>(update_id, transition_interval).first;
+        state_id_t state_id;
+        if (new_state) {
+            state_id = dm.create_state<gainState>(update_id, transition_interval).first;
+        }
+        // Use the current dataset ID
+        else {
+            const auto update = gains_fifo.get_update(double_to_ts(new_ts)).second;
```

Comment:
Oh, I see. There's always guaranteed to be an initial one.

Reply:
That's a good point actually. How is there always guaranteed to be an initial one?

Reply:
That's because `configUpdater` requires the static config to have initial values for all updatable config parts. When anything subscribes to an updatable part of the config, it receives an update callback with the initial values. 
Now that I'm thinking about it, this only works because you set the initial value of `new_state` in the config to `True`. Even though a false here doesn't make any sense, the code should be robust against a `False` there.